src/ui/client/src/components/Content.js

```diff
@@ -28,32 +28,24 @@ class Content extends React.Component {
     this.loadDestinationData();
   }
 
-  loadDestinationData = async e => {
-    if (e) e.preventDefault();
-
-    const response = await fetch(
-      "/api/v1/destinations/" + this.props.state.suggestion.city + "/" + this.props.state.suggestion.country
-    );
-
-    var data = await response.json();
-
-    console.log(data);
-
-    this.setState({ currentDestination: data });
-  };
+  async componentDidUpdate() {
+    if (this.state.currentDestination.city !== this.props.state.suggestion.city || this.state.currentDestination.country !== this.props.state.suggestion.country) {
+      this.loadDestinationData();
+    }
```

Comment:
```js
const { currentCity, currentCountry } = this.state.currentDestination
const { city, country } = this.props.state.suggestion

if (currentCity !== city || currentCountry !== country) {
  this.loadDestinationData();
}
```

Reply:
@MaxShapiro32 any update?

Reply:
@moficodes Just got back from vacation today. Going to take a look at this today

Reply:
@bourdakos1 I tried changing this if statement and I noticed that the page is much slower in rendering with your suggestion. I am not sure why this is the case.

Reply:
@MaxShapiro32 lies

Reply:
@bourdakos1 I figured out the issue. ` const { currentCity, currentCountry } = this.state.currentDestination` doesn't assign the correct values from `this.state.currentDestination` because there are more than 2 values in it. This is what it looks like:
```js
currentDestination: {
        city: "",
        lat: "",
        lng: "",
        country: "",
        population: "",
        description: ""
}
```
Since this is the case, I am going to break up that line into:
```js
    const currentCity = this.state.currentDestination.city;
    const currentCountry = this.state.currentDestination.country;
```

I have tried this and it works fine and is not slow.

Reply:
Oh yea, silly me, `currentCity` isn't part of the object